### DISCUSSION TITLE
What is the nativeless version?

### DISCUSSION THREAD

**Discussion by Jenix-Park** [melanchall/drywetmidi#207] [Q&A] [answered]
Hi,
Thanks for the new version!
By the way, if I remember correctly, this is my first time seeing "nativeless" versions.
So I'm confused about which to download.
Currently, all I do is read a MIDI file, modify it, and then write.
I don't access any MIDI devices using this plugin.

PS. Is there any breaking changes?

**Comment by melanchall** [comment, marked as answer]
Hi,

More info here: https://melanchall.github.io/drywetmidi/articles/dev/Nativeless-package.html. I recommend you to stick with the full version.

**Reply by Jenix-Park** [reply]
Hmm.. Reading the manual, for my current project, which is just a simple MIDI file reader/writer, the "nativeless" version seems just fine.
The application doesn't use any hardware and the playback feature.
But I'll download the full version as you said.

By the way, you didn't answer my last question!
Are there any breaking changes from the previous version?

**Reply by melanchall** [reply]
There are no breaking changes there. I always write them (if they exist) in release notes. As you can see in [release notes for 6.1.2](https://github.com/melanchall/drywetmidi/releases/tag/v6.1.2), it's a minor release without breaking changes. But, for example, in [release notes for 6.0.0](https://github.com/melanchall/drywetmidi/releases/tag/v6.0.0) there is a section describing such changes.

Yes, you can use nativeless version in your case if you want. But the main goal of this version is to cut out all native-dependent API if you have problems with it (for example, [in UWP apps](https://github.com/melanchall/drywetmidi/issues/108#issuecomment-1179123484)). But if you want to have a bit smaller app without redundant (in your case) binaries, maybe nativeless version is a good choice too :)
